features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Details/Comments/Comment.jsx

```diff
@@ -253,6 +254,146 @@ class Comment extends React.Component {
+                                },
+                            };
+                        }
+                        return [...acc, temp];
+                    }, []);
+                    if (commentsUpdate) {
+                        commentsUpdate(newComments);
+                    }
+                }
+            })
+            .catch((error) => {
+                if (process.env.NODE_ENV !== 'production') {
+                    console.log(error);
+                }
+            });
+    }
+
+    /**
+     * Delete reply
+     * @param {string} parentCommentId parent comment of reply
+     * @param {string} replyCommentId deleted reply comment
+     * @memberof Comments
+     */
+    handleDeleteReply(parentCommentId, replyCommentId) {
+        const { comments, updateComment, apiId } = this.props;
+        const existingComment = comments.find((item) => item.id === parentCommentId);
+        const { replies } = existingComment;
+        // updated values
+        const updatedRepliesList = replies.list.filter((reply) => reply.id !== replyCommentId);
+        const newTotal = replies.pagination.total - 1;
+        const newLimit = replies.pagination.limit > newTotal ? newTotal : replies.pagination.limit;
+        const newCount = replies.count - 1;
+
+        if (newTotal > newCount) {
+            const restApi = new API();
+            restApi
+                .getAllCommentReplies(apiId, parentCommentId, 1, newLimit - 1)
+                .then((result) => {
+                    const data = JSON.parse(result.data);
+                    if (data) {
+                        const updatedComment = {
+                            ...existingComment,
+                            replies: {
+                                ...replies,
+                                list: [...updatedRepliesList, ...data.list],
+                                pagination: {
+                                    ...replies.pagination,
+                                    total: newTotal,
+                                },
+                            },
+                        };
+                        if (updateComment) {
+                            updateComment(updatedComment);
+                        }
+                    }
+                })
+                .catch((error) => {
+                    if (process.env.NODE_ENV !== 'production') {
+                        console.log(error);
+                    }
+                });
+        } else {
+            const updatedComment = {
+                ...existingComment,
+                replies: {
+                    ...replies,
+                    count: newCount,
+                    list: updatedRepliesList,
+                    pagination: {
+                        ...replies.pagination,
+                        limit: newLimit,
+                        total: newTotal,
+                    },
+                },
+            };
+            if (updateComment) {
```

Comment:
From where do we get this `updateComment` prop and why we need such ?
I think we have to use [`getderivedstatefromprops`](https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops) since this is a class component

Reply:
`updateComment` is a method[1] in `Comments.jsx` introduced to replace the update functionality previously done using `updateCommentList()`. When a reply is deleted/added, this method update is used to update the corresponding comment in the existing comment list. This is passed as a prop to `Comment.jsx`. 

We can't use `getDerivedStateFromProps` in this scenario since we are updating a state in the parent component using a prop method right?



1. https://github.com/wso2/carbon-apimgt/pull/10226/files#diff-f026e14a38b0d47f4d5dbaf7788e9d7200d1f8db9aabe35a84a221a25847079dR231
2. https://github.com/wso2/carbon-apimgt/pull/10226/files#diff-f026e14a38b0d47f4d5dbaf7788e9d7200d1f8db9aabe35a84a221a25847079dR481

Reply:
Thanks , Yes let keep it simple like this 👍